### ISSUE TITLE
Django namespace class decorator magic and intent

### ISSUE THREAD

**Issue by daemonburrito** [abourget/gevent-socketio#118]
I have been working on using gevent-socketio for an existing django application. At first, using the latest version in pypi, I hand-rolled integration; but then, I looked closer at 0.3.6 and the django_chat app, and I found the more elegant solution in gevent.socketio.sdjango.

I am having trouble with using it, though. I think the problem is mine; django_chat works fine, even with a "backported" version of gevent.sdjango (urls.defaults for django 1.3).

So the example works fine, and my use doesn't, somehow. I'm getting no_such_namespace, though there is a socket connection. Inspecting the socket.io client, it is also configured with the correct namespace.

I think the gap is my understanding of the namespace class decorator magic and its intent. I've scoured and grep'd the django_chat example to death, and it appears that the mere act of importing the ChatNamespace registers it with the SOCKETIO_NS global.

Is this not correct? What am I missing?

tl;dr: Explanation for gevent.socketio.sdjango needed.

Will volunteer to write docs with answer.
